Makefile

```diff
@@ -1,83 +1,87 @@
-# Convenience data so we can run the following and include
-# sources from three directories deep.
-#
-# Example:
-# $(subst dir, emily/cdk/lib, $(THREE_DIRS_DEEP))
-# becomes
-# emily/cdk/lib/*  emily/cdk/lib/*/*  emily/cdk/lib/*/*/*
-#
-# TODO(TBD): Find a better way to do this.
-ONE_DIR_DEEP    := dir/*
-TWO_DIRS_DEEP   := dir/* $(subst dir, dir/*, $(ONE_DIR_DEEP))
-THREE_DIRS_DEEP := dir/* $(subst dir, dir/*, $(TWO_DIRS_DEEP))
-FOUR_DIRS_DEEP  := dir/* $(subst dir, dir/*, $(THREE_DIRS_DEEP))
-FIVE_DIRS_DEEP  := dir/* $(subst dir, dir/*, $(FOUR_DIRS_DEEP))
+# The absolute path to the directory containing the Makefile.
+WORKING_DIR := $(realpath $(shell dirname $(firstword $(MAKEFILE_LIST))))
 
 # Common Folders.
-AUTOGENERATED_SOURCES := ./.generated-sources
-
-# Blocklist Client Files
-AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT := $(AUTOGENERATED_SOURCES)/blocklist-api/src/lib.rs
-BLOCKLIST_OPENAPI_PATH := $(AUTOGENERATED_SOURCES)/blocklist-openapi-gen
-BLOCKLIST_OPENAPI_SPEC := $(BLOCKLIST_OPENAPI_PATH)/blocklist-client-openapi.json
-
-# Emily API Files
-EMILY_OPENAPI_PATH := $(AUTOGENERATED_SOURCES)/emily/openapi
-EMILY_OPENAPI_SPEC := $(EMILY_OPENAPI_PATH)/emily-openapi-spec.json
-AUTOGENERATED_EMILY_CLIENT := $(AUTOGENERATED_SOURCES)/emily/client/rust/src/lib.rs
-EMILY_LAMBDA_BINARY := target/lambda/emily-handler/bootstrap.zip
-EMILY_CDK_TEMPLATE := emily/cdk/cdk.out/EmilyStack.template.json
+AUTOGENERATED_SOURCES := .generated-sources
 
 # Don't use the install target here so you can rerun install without
 # Makefile complaints.
 export DATABASE_URL=postgres://user:password@localhost:5432/signer
 
+# The package flags for cargo commands.
+CARGO_FLAGS := --workspace --exclude emily-openapi-spec --exclude blocklist-openapi-gen
+
+# ##############################################################################
+# MAIN TARGETS
+# ##############################################################################
+
 install:
 	pnpm install
 
-build: $(AUTOGENERATED_BLOCKLIST_CLIENT_CLIENT) $(AUTOGENERATED_EMILY_CLIENT)
-	cargo build ${CARGO_BUILD_ARGS}
-	pnpm --recursive build
-	# TODO(719): No need to do this once rustfmt 2.0.0 ships
-	# Format generated sources:
-	cargo fmt -p emily-openapi-spec -p emily-client -p blocklist-api -p blocklist-openapi-gen
+build: blocklist-client-codegen emily-client-codegen contracts
+	cargo build --all-targets $(CARGO_FLAGS) ${CARGO_BUILD_ARGS}
```

Comment:
Do we want all targets here? From the [docs](https://doc.rust-lang.org/cargo/commands/cargo-build.html#target-selection), this is equivalent to `--lib --bins --tests --benches --examples`, which seems like more than we'd usually want. Maybe just the default `cargo build` would be better and you can supply the build arg with `CARGO_BUILD_ARGS=--all-targets make build`, if you want all targets. Thoughts?

Reply:
Tbh i was on the fence about removing this altogether because it's so ambiguous and we don't use it in any of our actual build processes.

`cargo build` will default build dev profile without the `--tests` flag, which is less than useful on a dev machine and not optimal for a release build. `--all-targets` gives us `--bin --lib --tests` which feels more relevant (we don't have examples or benches so those are moot).

Now we have separate `make contracts` etc. so `make build` isn't needed for codegen, and our other rules will spec/codegen if (and only if) the sources are outdated now.

So, I honestly don't know what's best 🤷🏻‍♂️ I never actually run this personally (except before when contract stuff was outdated).

Reply:
I don't use it either (at least not anymore). It is something that I've told people to do to get up and running for development, since it was used to build generated sources and smart contract stuff. But the part that seems unnecessary is `--bin`, which as time goes on will probably be even more unnecessary.

I don't actually feel strongly about it. I use `cargo build` and `pnpm build` when I need to.

I'll resolve this and leave the final decision to you.

Reply:
I'll leave it for now and we can change it in another PR if we decide we want to, I don't have any strong opinions either